tools/samtools.wdl

```diff
@@ -108,90 +117,92 @@ task flagstat {
 
     input {
         File bam
-        String outfile_name = basename(bam, ".bam")+".flagstat.txt"
-        Int max_retries = 1
+        String outfile_name = basename(bam, ".bam") + ".flagstat.txt"
         Int memory_gb = 5
+        Int modify_disk_size_gb = 0
+        Int max_retries = 1
     }
 
     Float bam_size = size(bam, "GiB")
-    Int disk_size = ceil((bam_size * 2) + 10)
+    Int disk_size_gb = ceil((bam_size * 1.2) + 10) + modify_disk_size_gb
 
-    command {
+    command <<<
         samtools flagstat ~{bam} > ~{outfile_name}
-    }
+    >>>
 
     output { 
        File flagstat_report = outfile_name
     }
 
     runtime {
-        disk: disk_size + " GB"
-        docker: 'quay.io/biocontainers/samtools:1.16.1--h6899075_1'
         memory: memory_gb + " GB"
+        disk: disk_size_gb + " GB"
+        docker: 'quay.io/biocontainers/samtools:1.16.1--h6899075_1'
         maxRetries: max_retries
     }
 }
 
 task index {
+    meta {
+        description: "This WDL task runs Samtools flagstat on the input BAM file. Produces statistics about the alignments based on the bit flags set in the BAM."
+    }
+
+    parameter_meta {
+        bam: "Input BAM format file to index"
+    }
+
     input {
         File bam
+        String outfile_name = basename(bam) + ".bai"  # TODO we might want to disable customizing the name for the BAI? Breaks our best practices, but also please do not change the BAI name
+        Boolean use_all_cores = false
+        Int memory_gb = 15  # TODO this seems like more RAM than should be necessary
+        Int modify_disk_size_gb = 0
```

Comment:
Do you have thoughts on either comment here?

I am in favor of letting the user customize these runs. But I think I'm more in favor of "please don't rename your bam index". I'd vote to remove the `outfile_name`/`prefix` parameter.